src/backend/expungeservice/models/charge_classifier.py

```diff
@@ -0,0 +1,118 @@
+from dataclasses import dataclass
+
+from expungeservice.models.case import Case
+
+
+@dataclass
+class ChargeClassifier:
+    case : Case
+    statute : str
+    level : str
+    chapter : str
+    section : str
+
+    def classify(self):
+        def classification_found(c):
+            return c is not None
+
+        for c in self.__classifications_list():
+            if classification_found(c):
+                return c
+
+    def __classifications_list(self):
+        yield ChargeClassifier._juvenile_charge(self.case)
+        yield from ChargeClassifier._set_classification_by_statute(self.statute, self.chapter, self.section)
+        yield ChargeClassifier._municipal_parking(self.case)
+        yield from ChargeClassifier._set_classification_by_level(self.level)
+        yield "UnclassifiedCharge"
+
+    @staticmethod
+    def _juvenile_charge(case):
+        if 'juvenile' in case.violation_type.lower():
+            return 'JuvenileCharge'
+
+    @staticmethod
+    def _set_classification_by_statute(statute, chapter, section):
+        yield ChargeClassifier._marijuana_ineligible(statute, section)
+        yield ChargeClassifier._list_b(section)
+        yield ChargeClassifier._crime_against_person(section)
+        yield ChargeClassifier._traffic_crime(statute)
+        yield ChargeClassifier._parking_ticket(statute, chapter)
+        yield ChargeClassifier._schedule_1_pcs(section)
+
+    @staticmethod
+    def _set_classification_by_level(level):
+        yield ChargeClassifier._non_traffic_violation(level)
+        yield ChargeClassifier._misdemeanor(level)
+        yield ChargeClassifier._felony_class_c(level)
+        yield ChargeClassifier._felony_class_b(level)
+        yield ChargeClassifier._felony_class_a(level)
+
+    @classmethod
+    def _marijuana_ineligible(cls, statute, section):
```

Comment:
Perhaps we can change all these methods to staticmethods since the class var is no longer needed?